README.md

```diff
@@ -406,16 +406,26 @@ where:
 
 By default, suggestions are rendered when the input isn't blank. Feel free to override this behaviour.
 
-This function gets the current value of the input, and it should return a boolean.
+This function gets the current value of the input and the reason why the suggestions might be rendered, and it should return a boolean.
 
 For example, to display suggestions only when input value is at least 3 characters long, do:
 
 ```js
-function shouldRenderSuggestions(value) {
+function shouldRenderSuggestions(value, reason) {
   return value.trim().length > 2;
 }
 ```
 
+You can use the second `reason` argument to finely control exactly when the suggestions are rendered. The possible values are closely related to those for `onSuggestionsFetchRequested`, plus a few extra cases:
+
+- `'input-changed'` - user typed something
+- `'input-focused'` - input was focused
+- `'input-blurred'` - input was un-focused
+- `'escape-pressed'` - user pressed <kbd>Escape</kbd> to clear the input (and suggestions are shown for empty input)
+- `'suggestions-revealed'` - user pressed <kbd>Up</kbd> or <kbd>Down</kbd> to reveal suggestions
```

Comment:
Not sure `suggestions-revealed` is the best event name actually... I mean, when `shouldRenderSuggestions` gets this, the suggestions are not yet revealed... Perhaps something like `arrow-pressed` is more honest? (Similar to `escape-pressed`).

Reply:
Yeah, I'm not 100% either way here... It's mostly for consistency, because that's the reason used in fetchRequested for the same event.

I'm not sure `arrow-pressed` would be perfectly honest either, as it isn't called every time an arrow is pressed, only when the suggestions are hidden beforehand, and an arrow is pressed to reveal them.